sos/cleaner/mappings/hostname_map.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2020 Red Hat, Inc. Jake Hunsaker <jhunsake@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+import re
+
+from sos.cleaner.mappings import SoSMap
+
+
+class SoSHostnameMap(SoSMap):
+    """Mapping store for hostnames and domain names
+
+    Hostnames are obfuscated using an incrementing counter based on the total
+    number of hosts matched regardless of domain name.
+
+    Domain names are obfuscated based on the host's hostname, plus any user
+    defined domains passed in by the `--domains` options.
+
+    Domains are obfuscated as whole units, meaning the domains 'example.com'
+    and 'host.foo.example.com' will be separately obfuscated with no relation
+    for example as 'obfuscatedomdain1.com' and 'obfuscatedomain2.com'.
+
+    Top-level domains are left untouched.
+    """
+
+    ignore_matches = [
+        'localhost',
+        '.*localdomain.*',
+        'com..*'
```

Comment:
Doesnt this ignore e.g. `www.mobil-com.cz` ?

Reply:
Good point, I'll change that to `^com.`. Although, the other consideration is to not skip this (e.g. com.redhat.access), but I couldn't find a good way to balance that with already-obfuscated domains.